src/main/java/spoon/support/reflect/cu/CompilationUnitImpl.java

```diff
@@ -301,6 +306,31 @@ public SourcePosition getOrCreatePartialSourcePosition() {
 		return myPartialSourcePosition;
 	}
 
+	@Override
+	public ElementSourceFragment getRootSourceFragment() {
+		if (rootFragment == null) {
+			String originSourceCode = getOriginalSourceCode();
```

Comment:
`getOriginalSourceCode()` might return null if `getFile()` returns null. You should check the result to avoid NPE here.